kqueue.go

```diff
@@ -266,16 +264,24 @@ func (w *Watcher) addWatch(name string, flags uint32) (string, error) {
 func (w *Watcher) readEvents() {
 	eventBuffer := make([]unix.Kevent_t, 10)
 
+	defer func() {
+		// cleanup
+		err := unix.Close(w.kq)
+		if err != nil {
+			select {
+			case w.Errors <- err:
+			case <-w.done:
```

Comment:
given the only way this function can return is if `w.done` is closed, I think this should instead be the default case.